error[E0510]: cannot assign `q` in match guard
  --> $DIR/match-guards-partially-borrow.rs:49:13
   |
LL |     match q {
   |           - value is immutable in match guard
...
LL |             q = true; //~ ERROR
   |             ^^^^^^^^ cannot assign
...
LL |         _ => (),
   |         - borrow later used here

error[E0510]: cannot assign `r` in match guard
  --> $DIR/match-guards-partially-borrow.rs:61:13
   |
LL |     match r {
   |           - value is immutable in match guard
...
LL |             r = true; //~ ERROR
   |             ^^^^^^^^ cannot assign
...
LL |         _ => (),
   |         - borrow later used here

error[E0503]: cannot use `s` because it was mutably borrowed
  --> $DIR/match-guards-partially-borrow.rs:70:11
   |
LL |     let h = &mut s;
   |             ------ borrow of `s` occurs here
LL |     match s { //~ ERROR
   |           ^ use of borrowed `s`
...
LL |             *h = !*h;
   |                   -- borrow later used here

error[E0510]: cannot assign `t` in match guard
  --> $DIR/match-guards-partially-borrow.rs:85:13
   |
LL |     match t {
   |           - value is immutable in match guard
...
LL |             t = true; //~ ERROR
   |             ^^^^^^^^ cannot assign
...
LL |         false => (),
   |         ----- borrow later used here

error[E0510]: cannot mutably borrow `x.0` in match guard
  --> $DIR/match-guards-partially-borrow.rs:99:22
   |
LL |     match x {
   |           - value is immutable in match guard
...
LL |                 Some(ref mut r) => *r = None, //~ ERROR
   |                      ^^^^^^^^^ cannot mutably borrow

error[E0510]: cannot assign `y` in match guard
  --> $DIR/match-guards-partially-borrow.rs:112:13
   |
LL |     match *y {
   |           -- value is immutable in match guard
...
LL |             y = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign
...
LL |         false => (),
   |         ----- borrow later used here

error[E0510]: cannot assign `z` in match guard
  --> $DIR/match-guards-partially-borrow.rs:123:13
   |
LL |     match z {
   |           - value is immutable in match guard
...
LL |             z = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign
...
LL |         &false => (),
   |         ------ borrow later used here

error[E0510]: cannot assign `a` in match guard
  --> $DIR/match-guards-partially-borrow.rs:135:13
   |
LL |     match a {
   |           - value is immutable in match guard
...
LL |             a = &true; //~ ERROR
   |             ^^^^^^^^^ cannot assign
...
LL |         false => (),
   |         ----- borrow later used here

error: aborting due to 8 previous errors

Some errors occurred: E0503, E0510.
For more information about an error, try `rustc --explain E0503`.
